sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/BulkWriter.scala

```diff
@@ -100,23 +104,26 @@ private class BulkWriter(container: CosmosAsyncContainer,
   private val errorCaptureFirstException = new AtomicReference[Throwable]()
   private val bulkInputEmitter: Sinks.Many[CosmosItemOperation] = Sinks.many().unicast().onBackpressureBuffer()
 
-  private val activeBulkWriteOperations = java.util.concurrent.ConcurrentHashMap.newKeySet[CosmosItemOperation]().asScala
-  private val activeReadManyOperations = java.util.concurrent.ConcurrentHashMap.newKeySet[ReadManyOperation]().asScala
+  private val activeBulkWriteOperations = mutable.LinkedHashSet[CosmosItemOperation]()
```

Comment:
does the linkedHashSet thread safe? as there might be cases where we need to add/remove item concurrently

Reply:
Good catch! (I actually checked - but misunderstood a comment in the API doc)  - Fixed